compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -724,28 +725,38 @@ public void visit(BLangPackage pkgNode) {
                 initFnBody.stmts.add(constInit);
             }
         }
+        List<BLangVariable> desugaredGlobalVarList = new ArrayList<>();
 
         pkgNode.globalVars.forEach(globalVar -> {
-            long globalVarFlags = globalVar.symbol.flags;
-            if (globalVar.expr != null || Symbols.isFlagOn(globalVarFlags, Flags.CONFIGURABLE)) {
-
+            // This will convert complex variables to simple variable
+            if (globalVar.getKind() == NodeKind.TUPLE_VARIABLE) {
+                BLangNode blockStatementNode = rewrite(globalVar, env);
+                // Add each desugared simple variable to global variables
+                ((BLangBlockStmt) blockStatementNode).stmts.forEach(bLangStatement -> {
+                    BLangSimpleVariableDef simpleVarDef = (BLangSimpleVariableDef) bLangStatement;
+                    addToInitFunction(simpleVarDef.var, initFnBody);
+                    desugaredGlobalVarList.add(rewrite(simpleVarDef.var, env));
+                });
+            } else {
```

Comment:
I think we can convert the `forEach` to a simple `for` loop and use `continue` at the end of the `if` block, and remove this `else` block and move the logic out.

Something like 

```
        for (BLangVariable globalVar : pkgNode.globalVars) {
            if (globalVar.getKind() == NodeKind.TUPLE_VARIABLE) {
                // ....   
                continue;
            }
            
            long globalVarFlags = globalVar.symbol.flags;
            // ...
        }
```

Reply:
okay

Reply:
Noted